unittests/test_utility.py

```diff
@@ -19,492 +17,547 @@
+    # Set `sys.ps1` to immitate an interactive session
+    monkeypatch.setattr(sys, 'ps1', 'rfm>>> ', raising=False)
+    assert os_ext.is_interactive()
+
+
+def test_is_url():
+    repo_https = 'https://github.com/eth-cscs/reframe.git'
+    repo_ssh = 'git@github.com:eth-cscs/reframe.git'
+    assert os_ext.is_url(repo_https)
+    assert not os_ext.is_url(repo_ssh)
+
+
+def test_git_repo_hash():
+    # A git branch hash consists of 8(short) or 40 characters.
+    assert len(os_ext.git_repo_hash()) == 8
+    assert len(os_ext.git_repo_hash(short=False)) == 40
+    assert os_ext.git_repo_hash(branch='invalid') is None
+    assert os_ext.git_repo_hash(branch='') is None
+
+
+def test_git_repo_exists():
+    assert os_ext.git_repo_exists('https://github.com/eth-cscs/reframe.git',
+                                  timeout=3)
+    assert not os_ext.git_repo_exists('reframe.git', timeout=3)
+    assert not os_ext.git_repo_exists('https://github.com/eth-cscs/xxx',
+                                      timeout=3)
+
+
+def test_force_remove_file(tmp_path):
+    fp = tmp_path / 'tmp_file'
+    fp.touch()
+    fp_name = str(fp)
+
+    assert os.path.exists(fp_name)
+    os_ext.force_remove_file(fp_name)
+    assert not os.path.exists(fp_name)
+
+    # Try to remove a non-existent file
+    os_ext.force_remove_file(fp_name)
+
+
+def test_expandvars_dollar():
+    text = 'Hello, $(echo World)'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    # Test nested expansion
+    text = '$(echo Hello, $(echo World))'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_backticks():
+    text = 'Hello, `echo World`'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    # Test nested expansion
+    text = '`echo Hello, `echo World``'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_mixed_syntax():
+    text = '`echo Hello, $(echo World)`'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    text = '$(echo Hello, `echo World`)'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+def test_expandvars_error():
+    text = 'Hello, $(foo)'
+    with pytest.raises(SpawnedProcessError):
+        os_ext.expandvars(text)
+
+
+def test_strange_syntax():
+    text = 'Hello, $(foo`'
+    assert 'Hello, $(foo`' == os_ext.expandvars(text)
+
+    text = 'Hello, `foo)'
+    assert 'Hello, `foo)' == os_ext.expandvars(text)
+
+
+def test_expandvars_nocmd(monkeypatch):
+    monkeypatch.setenv('FOO', 'World')
+    text = 'Hello, $FOO'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+    text = 'Hello, ${FOO}'
+    assert 'Hello, World' == os_ext.expandvars(text)
+
+
+@pytest.fixture
+def prefix_abs(tmp_path):
+    prefix = tmp_path / 'prefix'
+    prefix.mkdir()
+    return prefix.resolve()
+
+
+@pytest.fixture
+def target_abs(tmp_path):
+    target = tmp_path / 'target'
+    target.mkdir()
+    return target.resolve()
+
+
+@pytest.fixture
+def testfiles(prefix_abs, target_abs):
```

Comment:
As before, combine these two fixtures in one taking the prefix as an argument.

Reply:
The target_abs is uses also in many test functions below

Reply:
I fixed that.